sdk/opa.go

```diff
@@ -298,6 +298,96 @@ func newDecisionResult() (*DecisionResult, error) {
 	return result, nil
 }
 
+// Partial returns a named decision. This function is threadsafe.
+func (opa *OPA) Partial(ctx context.Context, options PartialOptions) (*DecisionResult, error) {
```

Comment:
If there's some way we can avoid duplicating all of the server/boilerplate, that would be nice.

Reply:
I'll try refactoring this

Reply:
I took a shot at refactoring.  Still trying to come up to speed with go, so not sure if it's the "go" way. Let me know if you see a better way to refactor.